cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -444,17 +454,56 @@ func (m *asgCache) createPlaceholdersForDesiredNonStartedInstances(groups []*aut
 
 		klog.V(4).Infof("Instance group %s has only %d instances created while requested count is %d. "+
 			"Creating placeholder instances.", *g.AutoScalingGroupName, realInstances, desired)
+
+		healthStatus := ""
+		isAvailable, err := m.isNodeGroupAvailable(g)
+		if err != nil {
+			klog.V(4).Infof("Could not check instance availability, creating placeholder node anyways: %v", err)
+		} else if !isAvailable {
+			klog.Warningf("Instance group %s cannot provision any more nodes!", *g.AutoScalingGroupName)
+			healthStatus = placeholderUnfulfillableStatus
+		}
+
 		for i := realInstances; i < desired; i++ {
 			id := fmt.Sprintf("%s-%s-%d", placeholderInstanceNamePrefix, *g.AutoScalingGroupName, i)
 			g.Instances = append(g.Instances, &autoscaling.Instance{
 				InstanceId:       &id,
 				AvailabilityZone: g.AvailabilityZones[0],
+				HealthStatus:     &healthStatus,
 			})
 		}
 	}
 	return groups
 }
 
+func (m *asgCache) isNodeGroupAvailable(group *autoscaling.Group) (bool, error) {
+	input := &autoscaling.DescribeScalingActivitiesInput{
+		AutoScalingGroupName: group.AutoScalingGroupName,
+		MaxRecords:           aws.Int64(1), // We only care about the most recent event
+	}
+
+	start := time.Now()
+	response, err := m.awsService.DescribeScalingActivities(input)
+	observeAWSRequest("DescribeScalingActivities", err, start)
+	if err != nil {
+		return true, err // If we can't describe the scaling activities we assume the node group is available
+	}
+
+	if len(response.Activities) > 0 {
+		activity := response.Activities[0]
```

Comment:
Do we only want to check the latest activity? Doesn't this potentially risk missing activities that have happened since the last time the ASG cache was updated if (for instance) an instance has been successfully taken out of service due to an instance rebalance recommendation?

Reply:
Ahh good point.  Let me see what happens here/if there's an easy way to resolve it.

Reply:
I just turned this into a loop that iterates over all the scaling activities since the ASG was updated, and early-aborts if any of them failed.  

I'm a _little_ bit worried about this approach, for example if something that wasn't a scale-up attempt failed -- but the only way I can think of to work around this would be text-matching against the `StatusMessage` response from DescribeScalingActivities, which feels like it could be super-brittle if, for example, AWS decides to change that text slightly.

@gjtempleton WDYT?

Reply:
I'm just an eager potential user of this feature but from my perspective it doesn't really matter why an ASG had a failure -- the ASG is still unhealthy. As long as you're checking, as you said, since the ASG cache was updated I think that it's going to be a win. If the error was `InsufficientInstanceCapacity` then it's the target situation but if it's something like `InstanceLimitExceeded` or a bad image / launch template misconfiguration then honestly the ASG is unhealthy and it should be marked as such. Ideally the reason would be logged but that's just a bonus for me. The main problem is AWS doesn't have a list of potential `Failure` scenarios but I think a failure is just a failure?

Thank you for making this!

Reply:
Yes, I think I agree with you.  The only case where I would be worried is if there was a Failure on scaling down, but it could still somehow scale up?  I don't even know if that's a thing that can happen.

Anyways, logging the failed scaling activity is a great suggestion, I added that.